react/common/components/ReturnDetails/RequestCancellation.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-console */
+import React from 'react'
+import type { FormEvent, ReactElement } from 'react'
+import { utils, Button, EXPERIMENTAL_Modal as Modal } from 'vtex.styleguide'
+import { useRuntime } from 'vtex.render-runtime'
+import { defineMessages, FormattedMessage } from 'react-intl'
+
+import { useReturnDetails } from '../../hooks/useReturnDetails'
+import { useUpdateRequestStatus } from '../../../admin/hooks/useUpdateRequestStatus'
+
+type CancellationMessage =
+  | 'adminAllow'
+  | 'adminRefuse'
+  | 'storeAllow'
+  | 'storeRefuse'
+
+export const messages = defineMessages({
+  adminAllow: {
+    id: 'return-app.return-request-details.cancellation.modal.adminAllow',
+  },
+  adminRefuse: {
+    id: 'return-app.return-request-details.cancellation.modal.adminRefuse',
+  },
+  storeAllow: {
+    id: 'return-app.return-request-details.cancellation.modal.storeAllow',
+  },
+  storeRefuse: {
+    id: 'return-app.return-request-details.cancellation.modal.storeRefuse',
+  },
+})
+
+const ParagraphChunk = (chunks: ReactElement) => <p>{chunks}</p>
+
+const RequestCancellation = () => {
+  const { isOpen, onOpen, onClose } = utils.useDisclosure()
+  const { data } = useReturnDetails()
+  const {
+    route: { domain },
+  } = useRuntime()
+
+  const { submitting, handleStatusUpdate } = useUpdateRequestStatus()
+
+  if (!data) return null
```

Comment:
Maybe I could replace this with the skeleton loader

Reply:
I think it's fine like that.